arch/simddetect.cpp

```diff
@@ -26,7 +26,7 @@
 #endif // x86 target
 
 #if defined(X86_BUILD)
-# if defined(__linux__) || defined(__MINGW32__) || defined(__APPLE__)
+# if defined(__GNUC__) || defined(__clang__) || defined(__MINGW32__) || defined(__MINGW64__) 
```

Comment:
Personally, I don't like redundancy in my code.

My point was that if `__GNUC__` does not have the meaning you thought it has.
It does not mean 'compiled with GCC'.

https://sourceforge.net/p/predef/wiki/Compilers